Microsoft.Alm.Authentication/Secret.cs

```diff
@@ -42,6 +42,18 @@ public static string UriToName(TargetUri targetUri, string @namespace)
             return targetName;
         }
 
+        public static string UriToUrl(TargetUri targetUri, string @namespace)
+        {
+            BaseSecureStore.ValidateTargetUri(targetUri);
+            if (String.IsNullOrWhiteSpace(@namespace))
+                throw new ArgumentNullException(@namespace);
+
+            string targetName = $"{@namespace}:{targetUri.ToString(false, true, true)}";
```

Comment:
The problem for Bitbucket it needs the username in the Url for per account credentials. 
AFAICS this extends the credential manager to include port and query path but not the username.
In effect Bitbucket would require ToString(true, false, false)